core/database/foxx/api/user_router.js

```diff
@@ -516,9 +519,54 @@ router.get('/token/set', function(req, res) {
                         refresh: req.queryParams.refresh,
                         expiration: Math.floor(Date.now() / 1000) + req.queryParams.expires_in
                     };
-                    g_db._update(user_id, obj, {
-                        keepNull: false
-                    });
+                    if (token_type && other_token_data) {
+                        // We need the '_key' from the user document, which is not necessarily a part of the above checks.
+                        const user_doc = g_db.u.document(user_id);
```

Comment:
I believe you should do this in the if else above, you already have the user document if it is the client, in that case you can avoid making another call to the database, you will still have to make the call if the user_id is the subject.

```
                    let user_doc = null;
                    if (req.queryParams.subject) {
                        user_id = req.queryParams.subject;
                        if (!g_db.u.exists(user_id))
                            throw [g_lib.ERR_INVALID_PARAM, "No such user '" + user_id + "'"];
                        g_lib.ensureAdminPermUser(client, user_id);
                        user_doc = g_db.u.document(user_id);
                    } else {
                        user_id = client._id;
                        user_doc = client;
                    }
```